sr_utilities_common/src/sr_utilities_common/aws_manager.py

```diff
@@ -71,66 +68,189 @@ def get_bucket_structure_with_prefix(self, bucket_name, prefix):
+                self.aws_paths.append(f"{files_folder_path}/{file_name}")
+
+    def _prepare_structure_download(self, files_base_path, files_folder_path, file_names, bucket_subfolder):
         self.file_full_paths = []
         self.aws_paths = []
         for file_name in file_names:
-            self.file_full_paths.append("{}/{}/{}".format(files_base_path, files_folder_path, file_name))
-            self.aws_paths.append("{}/{}".format(files_folder_path, file_name))
+            self.file_full_paths.append(f"{files_base_path}/{files_folder_path}/{file_name}")
+            if bucket_subfolder:
+                self.aws_paths.append(f"{bucket_subfolder}/{file_name}")
+            else:
+                self.aws_paths.append(file_name)
 
-    def download(self, bucket_name, files_base_path, files_folder_path, file_names):
-        self._prepare_structure(bucket_name, files_base_path, files_folder_path, file_names)
-        directory = "{}/{}".format(files_base_path, files_folder_path)
+    def download(self, bucket_name, files_base_path, files_folder_path, file_names, bucket_subfolder):
+        self._prepare_structure_download(files_base_path, files_folder_path, file_names, bucket_subfolder)
+        downloadSucceded = False
+        directory = os.path.join(files_base_path, files_folder_path)
         if not os.path.exists(directory):
             os.makedirs(directory)
-        downloadSucceded = False
-        for self.file_full_path, self.aws_path in zip(self.file_full_paths, self.aws_paths):
+        for file_full_path, aws_path in zip(self.file_full_paths, self.aws_paths):
+            directory = os.path.dirname(file_full_path)
+            if not os.path.exists(directory):
+                os.makedirs(directory)
             try:
-                self._client.download_file(bucket_name, self.aws_path, self.file_full_path)
+                self._client.download_file(bucket_name, aws_path, file_full_path)
                 downloadSucceded = True
             except Exception as e:
                 rospy.logwarn("File download failed. " + str(e))
         return downloadSucceded
 
-    def upload(self, bucket_name, files_base_path, files_folder_path, file_names):
-        self._prepare_structure(bucket_name, files_base_path, files_folder_path, file_names)
+    def upload(self, bucket_name, files_base_path, files_folder_path, file_names, bucket_subfolder):
+        self._prepare_structure_upload(files_base_path, files_folder_path, file_names, bucket_subfolder)
         uploadSucceded = False
-        for self.file_full_path, self.aws_path in zip(self.file_full_paths, self.aws_paths):
+        for file_full_path, aws_path in zip(self.file_full_paths, self.aws_paths):
             try:
-                self._client.upload_file(self.file_full_path, bucket_name, self.aws_path)
+                self._client.upload_file(file_full_path, bucket_name, aws_path)
                 uploadSucceded = True
             except Exception as e:
```

Comment:
Too general Exception, try catching more specific - probably related to wrong path, wrong client setup, no connection